Ref #7991. Supported hardware (#129)
micro-ROS/micro-ROS.github.io: _docs/overview/hardware/index.md

permalink: /docs/overview/hardware/
---

> micro-ROS targets mid-range and high-performance 32-bits microcontrollers families. For now, most of the ports are based on the [STM32 series](https://www.st.com/en/microcontrollers-microprocessors/stm32-32-bit-arm-cortex-mcus.html) from ST. These kinds of MCU feature ARM Cortex-M processors with many peripherals such as GPIO, communication or coprocessors.  By default, micro-ROS uses [NuttX RTOS](https://nuttx.org/), but it also has ports for [FreeRTOS](https://www.freertos.org/) and [Zephyr](https://www.zephyrproject.org/). These RTOSes have a big variety of supported MCUs and development boards. The next list shows some of them:

Reviewer: ```suggestion
micro-ROS targets mid-range and high-performance 32-bits microcontrollers families. For now, most of the ports are based on the [STM32 series](https://www.st.com/en/microcontrollers-microprocessors/stm32-32-bit-arm-cortex-mcus.html) from ST. These kinds of MCU feature ARM Cortex-M processors with many peripherals such as GPIO, communication or coprocessors.  By default, micro-ROS uses [NuttX RTOS](https://nuttx.org/), but it also has ports for [FreeRTOS](https://www.freertos.org/) and [Zephyr](https://www.zephyrproject.org/). These RTOSes have a big variety of supported MCUs and development boards. The follwing list shows some of them:
```

+ MicroChip PIC32MX Family
+ Atmel SAMA5Dx
+ STM32F1/2/3/4 and STM32L1/4, that includes many Nucleo board support.
+ Renesas M16C/26
+ NXP/Freescale i.MX1/6 and i.MX RT
+ Silicon Labs EFM32 Gecko and Giant Gecko
+ XTENSA board, that includes ESP32
+ RISC-V boards

<!-- TODO (pablogs): Update this paragraph to a more generic porting guide link  -->

> Please check [this link](https://cwiki.apache.org/confluence/display/NUTTX/Supported+Platforms) to check the complete list and the status of each board. In case you are interested in porting new boards or MCUs, please check the [next link](https://cwiki.apache.org/confluence/display/NUTTX/Porting+Guide).

Reviewer: ```suggestion
Please check [this link](https://cwiki.apache.org/confluence/display/NUTTX/Supported+Platforms) to see the complete list and the status of each board. In case you are interested in porting new boards or MCUs, please check the [next link](https://cwiki.apache.org/confluence/display/NUTTX/Porting+Guide).
```

Even though many development boards could be used, we have chosen two of them as references. This page lists the hardware platforms that we use to test and develop micro-ROS, and also accessories that we frequently refer to, such as add-on boards, and JTAG probes.

> For the ease of use, micro-ROS provides a ready to use examples for some development boards. These out-of-the-box examples aim to show micro-ROS capabilities and they are also a starting point for developing embedded ROS 2 applications.

Reviewer: ```suggestion
For the ease of use, micro-ROS provides a ready to use example for some development boards. These out-of-the-box examples aim to show micro-ROS capabilities and they are also a starting point for developing embedded ROS 2 applications.
```

> It contains 196KB of RAM and 1MB of Flash. It is a very complete board thanks to the wide variety of communication interfaces it offers: USB OTG, Ethernet, SD Card slot, SPI, CAN or I2C buses are exposed. The board contains various expansion options available: Arduino-like headers for attaching daughter boards, many pins exposed, as well as a UEXT connector. This connector is a custom pin-out bus and is used to attach sensor breakouts sensors that manufacturer sells.

Reviewer: sensor breakout sensors?

Author: Done

Ports for micro-ROS on all supported RTOS are available for this board. Examples on how to start developing with this board are available:
 - [Zephyr](/docs/tutorials/advanced/zephyr/zephyr_getting_started/)
 - [Nuttx](/docs/tutorials/advanced/nuttx/nuttx_getting_started/)
 - [FreeRTOS](/docs/tutorials/advanced/freertos/freertos_getting_started/)

### Development tools

> In order to flash and debug the board, it is required to get a JTAG probe. We recommend getting on of the next JTAG probes:

Reviewer: ```suggestion
In order to flash and debug the board, it is required to get a JTAG probe. We recommend getting one of the following JTAG probes:
```

<img width="400" src="imgs/2.jpg">

The STM32L4 Discovery kit IoT enables a wide diversity of applications by exploiting low-power communication, multiway sensing and Arm Cortex M4 core-based STM32L4 Series features.
The support for Arduino and PMOD connectivity provides unlimited expansion capabilities with a large choice of specialized add-on boards.

This board features a STM32L475E MCU with 1 MB of Flash memory and 128 KB of RAM. In addition to the MCU peripherals, the board includes: 
 - 64 Mb SPI Flash memory
 - Bluetooth V4.1 module (SPBTLE-RF)
 - 915 MHz low-power RF module (SPSGRF-915)
 - 802.11 b/g/n module (ISM43362-M3G-L44)
 - NFC tag based on M24SR with printed antenna
 - 2 digital microphones (MP34DT01)
 - relative humidity and temperature digital sensor (HTS221)
 - 3-axis magnetometer (LIS3MDL)
 - 3-axis accelerometer and gyroscope (LSM6DSL)
 - digital barometer (LPS22HB)
 - Time-of-Flight and gesture-detection sensor (VL53L0X)
 - programmable push-buttons
 - USB OTG FS with Micro-AB connector
 - on-board ST-LINK/V2 debugger and programmer

> Examples on how to start developing with this board are available [here](/docs/tutorials/demos/tof_demo/).

Reviewer: link doesn't work
(other links are not working either)

Author: We cant use the links until ther are merged into [microros.github.io](microros.github.io)
